frontend/src/components/pages/CampCreation/index.tsx

```diff
@@ -159,87 +170,254 @@ const CampCreationPage = (): React.ReactElement => {
+          (fq) => {
+            return {
+              question: fq.question,
+              type: fq.type,
+              category: fq.category,
+              required: fq.required,
+              description: fq.description,
+              options: fq.options,
+            };
+          },
+        );
+        setCustomQuestions(currentFormQuestions);
+
+        // Set basic camp details
+        setCampName(editCamp.name);
+        setCampDescription(editCamp.description);
+        setDailyCampFee(editCamp.fee);
+        setStartTime(editCamp.startTime);
+        setEndTime(editCamp.endTime);
+        setAgeLower(editCamp.ageLower);
+        setAgeUpper(editCamp.ageUpper);
+        setOffersEDLP(
+          editCamp.earlyDropoff.length !== 0 ||
+            editCamp.latePickup.length !== 0,
+        );
+        setEarliestDropOffTime(editCamp.earlyDropoff);
+        setLatestPickUpTime(editCamp.latePickup);
+        setPriceEDLP(editCamp.pickupFee);
+        setAddresLine1(editCamp.location.streetAddress1);
+        setAddresLine2(editCamp.location.streetAddress2 ?? "");
+        setCity(editCamp.location.city);
+        setProvince(editCamp.location.province);
+        setPostalCode(editCamp.location.postalCode);
+        setCampImageURL(editCamp.campPhotoUrl);
+
+        // Set camp level capacity from camp sessions
+        if (editCamp.campSessions.length > 0) {
+          // At the creation stage, all the sessions should have the same capacity
+          setCampCapacity(editCamp.campSessions[0].capacity);
+        }
+        setIsLoadedCampData(true);
+      }
+    };
+
+    // We only want to fetch the current state of the camp if we are editing a draft camp
+    if (editCampId) {
+      setInitialEditState();
+    }
+  }, [editCampId]);
+
+  const createUpdateCampHelper = async (
+    isPublishedCamp: boolean,
+    isNewCamp: boolean,
+  ): Promise<void> => {
+    const campFields: CreateUpdateCampRequest = {
+      active: isPublishedCamp,
+      ageLower,
+      ageUpper,
+      campCoordinators: [],
+      campCounsellors: [],
+      name: campName,
+      description: campDescription,
+      earlyDropoff: earliestDropOffTime,
+      endTime,
+      latePickup: latestPickUpTime,
+      location: {
+        streetAddress1: addressLine1,
+        streetAddress2: addressLine2,
+        city,
+        province,
+        postalCode,
+      },
+      startTime,
+      fee: dailyCampFee,
+      pickupFee: priceEDLP,
+      dropoffFee: priceEDLP,
+      formQuestions: customQuestions,
+      campSessions: scheduledSessions.map((campSession) => {
+        return {
+          dates: campSession.dates.map((date) => date.toString()),
+          capacity: campCapacity,
+        };
+      }),
+      volunteers: "",
+    };
+
+    try {
+      const campResponse = await createUpdateCamp(
+        campFields,
+        isNewCamp,
+        editCampId,
+      );
+
+      if (campResponse) {
```

Comment:
Because we do: 
```ts
return error as CreateUpdateCampResponse;
```

I would guess this evaluates to true no matter success/error, can we check for `campResponse.id` or type instead? (I may be completely wrong)

Reply:
I just fixed the campsAPI functions for create and update to actually throw errors so we can catch them properly.